qcodes/instrument_drivers/Keysight/private/Keysight_344xxA_submodules.py

```diff
@@ -514,6 +637,36 @@ def __init__(self, name: str, address: str, silent: bool=False,
                            label='Voltage',
                            unit='V')
 
+        #####################################
+        # Time trace parameters
+
+        self.add_parameter('timetrace_npts',
+                           label='Time trace number of points',
+                           initial_value=500,
+                           get_cmd=None,
+                           set_cmd=None,
+                           vals=vals.Ints(1))
+
+        self.add_parameter('timetrace_dt',
+                           label='Time trace time interval',
+                           unit='s',
+                           initial_value=1e-1,
+                           get_cmd=None,
+                           set_cmd=None,
+                           vals=vals.Numbers(0))
+
+        self.add_parameter('time_axis',
+                           label='Time',
+                           unit='s',
+                           snapshot_value=False,
+                           vals=vals.Arrays(shape=(self.timetrace_npts,)),
+                           parameter_class=TimeAxis)
+
+        self.add_parameter('timetrace',
+                           vals=vals.Arrays(shape=(self.timetrace_npts,)),
+                           setpoints=(self.time_axis,),
```

Comment:
```suggestion
                           setpoints=(self.time_axis,),
                           snapshot_value=False,
```

Reply:
Happens automatically for a `ParameterWithSetpoint`. For reference:
https://github.com/QCoDeS/Qcodes/blob/9bf9494861897198660255ac69ff7bb498d8f9b0/qcodes/instrument/parameter.py#L1060-L1065